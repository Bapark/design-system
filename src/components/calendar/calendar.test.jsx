import React from 'react';
import {
  render,
  screen,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Calendar from './calendar.jsx';

describe('<Calendar />', () => {
  const requiredProps = {
  };

  it('has defaults', () => {
    jest.spyOn(Date, 'now').mockImplementation(() => 1596062788778); // July 29, 2020
    render(<Calendar {...requiredProps} />);
    expect(document.body).toMatchSnapshot();
  });

  describe('previous month button', () => {
    it('displays the previous month', () => {
      jest.spyOn(Date, 'now').mockImplementation(() => 1596062788778); // July 29, 2020
      render(<Calendar {...requiredProps} />);
      userEvent.click(screen.getByText('Prev'));
      expect(screen.getByText('June 2020')).toBeInTheDocument();
    });
  });

  describe('next month button', () => {
    it('displays the next month', () => {
      jest.spyOn(Date, 'now').mockImplementation(() => 1596062788778); // July 29, 2020
      render(<Calendar {...requiredProps} />);
      userEvent.click(screen.getByText('Next'));
      expect(screen.getByText('August 2020')).toBeInTheDocument();
    });
  });

  describe('value props API', () => {
    it('can be set', () => {
      expect(render(<Calendar {...requiredProps} value="2012-02-28" />));
      expect(screen.getByText('February 2012')).toBeInTheDocument();
    });
  });
});
