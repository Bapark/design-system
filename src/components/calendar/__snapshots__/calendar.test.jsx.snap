// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> has defaults 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <div
        class="header"
      >
        <svg
          class="change-month-button"
          height="8"
          role="button"
          tabindex="0"
          width="12"
        >
          <title>
            Change calendar to June 2020
          </title>
          <use
            xlink:href="#arrow-left.svg"
          />
        </svg>
        <button
          class="year-button"
        >
          July 2020
          <svg
            class="change-year-icon"
            height="6"
            role="img"
            width="10"
          >
            <use
              xlink:href="#caret-down.svg"
            />
          </svg>
        </button>
        <svg
          class="change-month-button"
          height="8"
          role="button"
          tabindex="0"
          width="12"
        >
          <title>
            Change calendar to August 2020
          </title>
          <use
            xlink:href="#arrow-right.svg"
          />
        </svg>
      </div>
      <table
        class="calendar-grid"
      >
        <thead>
          <tr>
            <th
              class="weekday"
            >
              S
            </th>
            <th
              class="weekday"
            >
              M
            </th>
            <th
              class="weekday"
            >
              T
            </th>
            <th
              class="weekday"
            >
              W
            </th>
            <th
              class="weekday"
            >
              T
            </th>
            <th
              class="weekday"
            >
              F
            </th>
            <th
              class="weekday"
            >
              S
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="not-current-date"
            >
              28
            </td>
            <td
              class="not-current-date"
            >
              29
            </td>
            <td
              class="not-current-date"
            >
              30
            </td>
            <td
              class="date"
            >
              1
            </td>
            <td
              class="date"
            >
              2
            </td>
            <td
              class="date"
            >
              3
            </td>
            <td
              class="date"
            >
              4
            </td>
          </tr>
          <tr>
            <td
              class="date"
            >
              5
            </td>
            <td
              class="date"
            >
              6
            </td>
            <td
              class="date"
            >
              7
            </td>
            <td
              class="date"
            >
              8
            </td>
            <td
              class="date"
            >
              9
            </td>
            <td
              class="date"
            >
              10
            </td>
            <td
              class="date"
            >
              11
            </td>
          </tr>
          <tr>
            <td
              class="date"
            >
              12
            </td>
            <td
              class="date"
            >
              13
            </td>
            <td
              class="date"
            >
              14
            </td>
            <td
              class="date"
            >
              15
            </td>
            <td
              class="date"
            >
              16
            </td>
            <td
              class="date"
            >
              17
            </td>
            <td
              class="date"
            >
              18
            </td>
          </tr>
          <tr>
            <td
              class="date"
            >
              19
            </td>
            <td
              class="date"
            >
              20
            </td>
            <td
              class="date"
            >
              21
            </td>
            <td
              class="date"
            >
              22
            </td>
            <td
              class="date"
            >
              23
            </td>
            <td
              class="date"
            >
              24
            </td>
            <td
              class="date"
            >
              25
            </td>
          </tr>
          <tr>
            <td
              class="date"
            >
              26
            </td>
            <td
              class="date"
            >
              27
            </td>
            <td
              class="date"
            >
              28
            </td>
            <td
              class="highlighted-date"
            >
              29
            </td>
            <td
              class="date"
            >
              30
            </td>
            <td
              class="date"
            >
              31
            </td>
            <td
              class="not-current-date"
            >
              1
            </td>
          </tr>
          <tr>
            <td
              class="not-current-date"
            >
              2
            </td>
            <td
              class="not-current-date"
            >
              3
            </td>
            <td
              class="not-current-date"
            >
              4
            </td>
            <td
              class="not-current-date"
            >
              5
            </td>
            <td
              class="not-current-date"
            >
              6
            </td>
            <td
              class="not-current-date"
            >
              7
            </td>
            <td
              class="not-current-date"
            >
              8
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</body>
`;

exports[`<Calendar /> renders the previous month 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <div
        class="header"
      >
        <svg
          class="change-month-button"
          height="8"
          role="button"
          tabindex="0"
          width="12"
        >
          <title>
            Change calendar to May 2020
          </title>
          <use
            xlink:href="#arrow-left.svg"
          />
        </svg>
        <button
          class="year-button"
        >
          June 2020
          <svg
            class="change-year-icon"
            height="6"
            role="img"
            width="10"
          >
            <use
              xlink:href="#caret-down.svg"
            />
          </svg>
        </button>
        <svg
          class="change-month-button"
          height="8"
          role="button"
          tabindex="0"
          width="12"
        >
          <title>
            Change calendar to July 2020
          </title>
          <use
            xlink:href="#arrow-right.svg"
          />
        </svg>
      </div>
      <table
        class="calendar-grid"
      >
        <thead>
          <tr>
            <th
              class="weekday"
            >
              S
            </th>
            <th
              class="weekday"
            >
              M
            </th>
            <th
              class="weekday"
            >
              T
            </th>
            <th
              class="weekday"
            >
              W
            </th>
            <th
              class="weekday"
            >
              T
            </th>
            <th
              class="weekday"
            >
              F
            </th>
            <th
              class="weekday"
            >
              S
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="not-current-date"
            >
              31
            </td>
            <td
              class="date"
            >
              1
            </td>
            <td
              class="date"
            >
              2
            </td>
            <td
              class="date"
            >
              3
            </td>
            <td
              class="date"
            >
              4
            </td>
            <td
              class="date"
            >
              5
            </td>
            <td
              class="date"
            >
              6
            </td>
          </tr>
          <tr>
            <td
              class="date"
            >
              7
            </td>
            <td
              class="date"
            >
              8
            </td>
            <td
              class="date"
            >
              9
            </td>
            <td
              class="date"
            >
              10
            </td>
            <td
              class="date"
            >
              11
            </td>
            <td
              class="date"
            >
              12
            </td>
            <td
              class="date"
            >
              13
            </td>
          </tr>
          <tr>
            <td
              class="date"
            >
              14
            </td>
            <td
              class="date"
            >
              15
            </td>
            <td
              class="date"
            >
              16
            </td>
            <td
              class="date"
            >
              17
            </td>
            <td
              class="date"
            >
              18
            </td>
            <td
              class="date"
            >
              19
            </td>
            <td
              class="date"
            >
              20
            </td>
          </tr>
          <tr>
            <td
              class="date"
            >
              21
            </td>
            <td
              class="date"
            >
              22
            </td>
            <td
              class="date"
            >
              23
            </td>
            <td
              class="date"
            >
              24
            </td>
            <td
              class="date"
            >
              25
            </td>
            <td
              class="date"
            >
              26
            </td>
            <td
              class="date"
            >
              27
            </td>
          </tr>
          <tr>
            <td
              class="date"
            >
              28
            </td>
            <td
              class="date"
            >
              29
            </td>
            <td
              class="date"
            >
              30
            </td>
            <td
              class="not-current-date"
            >
              1
            </td>
            <td
              class="not-current-date"
            >
              2
            </td>
            <td
              class="not-current-date"
            >
              3
            </td>
            <td
              class="not-current-date"
            >
              4
            </td>
          </tr>
          <tr>
            <td
              class="not-current-date"
            >
              5
            </td>
            <td
              class="not-current-date"
            >
              6
            </td>
            <td
              class="not-current-date"
            >
              7
            </td>
            <td
              class="not-current-date"
            >
              8
            </td>
            <td
              class="not-current-date"
            >
              9
            </td>
            <td
              class="not-current-date"
            >
              10
            </td>
            <td
              class="not-current-date"
            >
              11
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</body>
`;

exports[`<Calendar /> renders the previous year 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <div
        class="header"
      >
        <svg
          class="change-month-button"
          height="8"
          role="button"
          tabindex="0"
          width="12"
        >
          <title>
            Change calendar to June 2019
          </title>
          <use
            xlink:href="#arrow-left.svg"
          />
        </svg>
        <button
          class="year-button"
        >
          July 2019
          <svg
            class="change-year-icon"
            height="6"
            role="img"
            width="10"
          >
            <use
              xlink:href="#caret-down.svg"
            />
          </svg>
        </button>
        <svg
          class="change-month-button"
          height="8"
          role="button"
          tabindex="0"
          width="12"
        >
          <title>
            Change calendar to August 2019
          </title>
          <use
            xlink:href="#arrow-right.svg"
          />
        </svg>
      </div>
      <table
        class="calendar-grid"
      >
        <thead>
          <tr>
            <th
              class="weekday"
            >
              S
            </th>
            <th
              class="weekday"
            >
              M
            </th>
            <th
              class="weekday"
            >
              T
            </th>
            <th
              class="weekday"
            >
              W
            </th>
            <th
              class="weekday"
            >
              T
            </th>
            <th
              class="weekday"
            >
              F
            </th>
            <th
              class="weekday"
            >
              S
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="not-current-date"
            >
              30
            </td>
            <td
              class="date"
            >
              1
            </td>
            <td
              class="date"
            >
              2
            </td>
            <td
              class="date"
            >
              3
            </td>
            <td
              class="date"
            >
              4
            </td>
            <td
              class="date"
            >
              5
            </td>
            <td
              class="date"
            >
              6
            </td>
          </tr>
          <tr>
            <td
              class="date"
            >
              7
            </td>
            <td
              class="date"
            >
              8
            </td>
            <td
              class="date"
            >
              9
            </td>
            <td
              class="date"
            >
              10
            </td>
            <td
              class="date"
            >
              11
            </td>
            <td
              class="date"
            >
              12
            </td>
            <td
              class="date"
            >
              13
            </td>
          </tr>
          <tr>
            <td
              class="date"
            >
              14
            </td>
            <td
              class="date"
            >
              15
            </td>
            <td
              class="date"
            >
              16
            </td>
            <td
              class="date"
            >
              17
            </td>
            <td
              class="date"
            >
              18
            </td>
            <td
              class="date"
            >
              19
            </td>
            <td
              class="date"
            >
              20
            </td>
          </tr>
          <tr>
            <td
              class="date"
            >
              21
            </td>
            <td
              class="date"
            >
              22
            </td>
            <td
              class="date"
            >
              23
            </td>
            <td
              class="date"
            >
              24
            </td>
            <td
              class="date"
            >
              25
            </td>
            <td
              class="date"
            >
              26
            </td>
            <td
              class="date"
            >
              27
            </td>
          </tr>
          <tr>
            <td
              class="date"
            >
              28
            </td>
            <td
              class="date"
            >
              29
            </td>
            <td
              class="date"
            >
              30
            </td>
            <td
              class="date"
            >
              31
            </td>
            <td
              class="not-current-date"
            >
              1
            </td>
            <td
              class="not-current-date"
            >
              2
            </td>
            <td
              class="not-current-date"
            >
              3
            </td>
          </tr>
          <tr>
            <td
              class="not-current-date"
            >
              4
            </td>
            <td
              class="not-current-date"
            >
              5
            </td>
            <td
              class="not-current-date"
            >
              6
            </td>
            <td
              class="not-current-date"
            >
              7
            </td>
            <td
              class="not-current-date"
            >
              8
            </td>
            <td
              class="not-current-date"
            >
              9
            </td>
            <td
              class="not-current-date"
            >
              10
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</body>
`;
